### ISSUE TITLE
Debug assertion failed

### ISSUE THREAD

**Issue by MathewFlemmer** [carnegierobotics/LibMultiSense#94]
I've done a fresh install of VS2022 and CMAKE on a Windows 10 machine.

I used the CMAKE gui to generate the VS solution using all default options.

I then built the All Build and Install projects, and ran the PointCloudUtility from inside VS. I needed to copy the .dll from either the install directory or the source folder.

It crashes with: 
Debug Assertion Failed!
Program:
....sense\build\source\Utilities\PointCloudUtility\MultiSense.dll
File: C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.33.31629\include\deque
Line: 284

Expression: cannot increment deque iterator past end

I don't have this issue when I use release 4.0

**Comment by mattalvarado** [comment]
Thanks for report @MathewFlemmer. 

What version hash are you seeing this issue under? 

What  model of camera are you testing with?

**Comment by MathewFlemmer** [comment]
I'm using an s7.

I'm using the latest release when I get the issue, v4.2.1

**Comment by mattalvarado** [comment]
@MathewFlemmer could you provide more info on your setup?

- What steps did you take to checkout and built the version of LibMultiSense you are testing
- What is the model number of computer you are testing on?
- What does your network setup look like? Are you connecting directly to the MultiSense via a PCIe NIC on your machine, or are you using a USB->Ethernet dongle?
- What MTU are you using to communicate with the camera?
- What arguments are you using to run the PointCloudUtiliy?

**Comment by MathewFlemmer** [comment]
I installed vs22 with all c++ tools, and cmake. I then downloaded libmultisense v4.2.1, extracted and ran cmake with default settings. I then built the all build and install projects in the lib multisense solution. I then set pointcloud utility as the default startup project and ran it.

The computer is an HP ZBook 15u G6.

I'm plugged directly into the multisense s7 with cable shipped in the developer kit plugged into the PCIe NIC card on my laptop.

I've left the MTU at 7200.

I'm running the PointCloudUtility with no arguments, just setting it as the default startup project and hitting F5 in VS.
